docs/reference/commands.rst

```diff
@@ -41,16 +62,144 @@ upgrade phase. Refer to the output below for a description of all available opti
     Usage: cou upgrade [options]
 
     Run the cloud upgrade.
+    If upgrade-group is unspecified, upgrade the whole cloud.
+
+    Options:
+      -h, --help            Show this help message and exit.
+      --model MODEL_NAME    Set the model to operate on.
+                            If not set, the currently active Juju model will be used.
+      --backup, --no-backup
+                            Include database backup step before cloud upgrade.
+                            Default to enabling database backup.
+      --force               Force the plan/upgrade of non-empty hypervisors.
+      --verbose, -v         Increase logging verbosity in STDOUT.
+                            Multiple 'v's yield progressively more detail (up to 4).
+                            Note that by default the logfile will include standard logs
+                            from juju and websockets, as well as debug logs from all other
+                            modules. To also include the debug level logs from juju and
+                            websockets modules, use the maximum verbosity.
+      --quiet, -q           Disable output in STDOUT.
+      --auto-approve        Automatically approve and continue with each upgrade step without prompt.
+
+    Upgrade group:
+      {control-plane,data-plane,hypervisors}
+                            Run 'cou upgrade <upgrade-group> -h' for more info about an upgrade group
+        control-plane       Run upgrade for the control-plane components.
+        data-plane          Upgrade all data-plane components.
+                            This is possible only if control-plane has been fully upgraded,
+                            otherwise an error will be thrown.
+        hypervisors         Upgrade machines with nova-compute and colocated services.
+                            This is possible only if control-plane has been fully upgraded,
+                            otherwise an error will be thrown.
+
+By default COU upgrades the entire OpenStack cloud with `cou upgrade`. However, the upgrade
+process can be tailored to target a specific group through a sub-command for more granular
+control. For further details, please see the `Upgrade Groups`_ section.
+
+Upgrade Groups
+--------------
+
+In COU, users can choose to selectively target only certain components in OpenStack cloud
+for upgrades, grouped by their roles. The available upgrade groups are **control-plane**,
+**data-plane**, and **hypervisors**.
```

Comment:
nit:
Maybe writing somewhere that this upgrade groups can be used for both `plan` and `upgrade`will make it clear. There is a link on commands for plan pointing here, but all the examples are with upgrade.